### ISSUE TITLE
Python Linux Wrapping Build Warning

### ISSUE THREAD

**Issue by tbirdso** [InsightSoftwareConsortium/ITK#3433]
### Description

I am encountering a build error when trying to update my local build from the ITK `master` branch on Linux:

### Steps to Reproduce

1. Update ITK to latest with `git checkout master; git pull`
2. Build with `cmake --build .`

### Expected behavior

Build succeeds

### Actual behavior

```none
[ 47%] Built target ITKTestKernel-all
make[2]: *** No rule to make target 'Wrapping/Generators/Python/itk-pkl/itkPyImageFilter.index.txt', needed by 'Wrapping/CMakeFiles/itk-stub-files'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:15467: Wrapping/CMakeFiles/itk-stub-files.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

### Reproducibility

100%

### Versions

`master`

### Environment

Ubuntu 18.04 LTS
CMake 3.21.3
Build generator GNU Make 4.2.1

### Additional Information

I am using a custom CMake configuration with some Python modules disabled to improve local build time. Perhaps some of these should be enforced as enabled?

```
ITK_BUILD_DEFAULT_MODULES=OFF
ITKGroup_Bridge=OFF
ITKGroup_Compatibility=OFF
ITKGroup_Core=ON
ITKGroup_Filtering=ON
ITKGroup_IO=ON
ITKGroup_Nonunit=OFF
ITKGroup_Numerics=ON
ITKGroup_Registration=OFF
ITKGroup_Remote=OFF
ITKGroup_Segmentation=OFF
ITKGroup_ThirdParty=OFF
ITKGroup_Video=OFF
```

**Comment by thewtex** [comment]
I also experienced another build error when attempting to reproduce the Windows Python CI issue locally with ITK_BUILD_DEFAULT_MODULES OFF. There is likely a build output dependency issue.

Is your build with the Ninja generator?

**Comment by tbirdso** [comment]
@thewtex I double checked and my build is using the default Unix Makefiles generator, I have updated the description. Is there a list of officially supported generators for ITK?

**Comment by thewtex** [comment]
Both should work.

It is good to know that you are getting this on Unix Makefiles. This is less strict than Ninja, which requires `BYPRODUCTS` to be set.


Re #3425 I suspect

https://github.com/InsightSoftwareConsortium/ITK/blob/7e5f6c703539c519dd0e3bd24a3bd2fc6d13ea8d/Wrapping/macro_files/itk_end_wrap_module.cmake#L119

the `BYPRODUCTS` should be elevated to `OUTPUT` or something similar.

The same here for this issue:

https://github.com/InsightSoftwareConsortium/ITK/blob/7e5f6c703539c519dd0e3bd24a3bd2fc6d13ea8d/Wrapping/macro_files/itk_end_wrap_module.cmake#L119

?

@tbirdso does this address your build?


@hjmjohnson does this make sense? Any thoughts?

**Comment by tbirdso** [comment]
I am seeing the same error after moving `${ITK_PYI_INDEX_FILES}` from `BYPRODUCTS` to `OUTPUTS`.

**Comment by hubutui** [comment]
I also get the same error, see the complete build log [here](https://github.com/InsightSoftwareConsortium/ITK/files/8721827/itk-buildlog.txt). You could skip the environment step and read it from line 2082, and check the error message from line 4763. This pkg is build with this [PKGBUILD](https://github.com/archlinuxcn/repo/blob/69559eaba3dbd330c5d546fafb96b80404c81850/archlinuxcn/itk-git/PKGBUILD), inshort:
```text
  cmake_opts=(
    -DBUILD_SHARED_LIBS=ON
    -DBUILD_TESTING=OFF
    -DCMAKE_BUILD_TYPE=Release
    -DCMAKE_INSTALL_PREFIX=/usr
    -DCMAKE_SKIP_INSTALL_RPATH=ON
    -DITK_BUILD_DEFAULT_MODULES=ON
    -DITK_USE_MKL=ON
    -DITK_USE_SYSTEM_CASTXML=ON
    -DITK_USE_SYSTEM_DCMTK=ON
    -DITK_USE_SYSTEM_DOUBLECONVERSION=OFF
    -DITK_USE_SYSTEM_EIGEN=ON
    -DITK_USE_SYSTEM_EXPAT=ON
    -DITK_USE_SYSTEM_FFTW=ON
    -DITK_USE_SYSTEM_GDCM=ON
    -DITK_USE_SYSTEM_GOOGLETEST=ON
    -DITK_USE_SYSTEM_HDF5=ON
    -DITK_USE_SYSTEM_JPEG=ON
    -DITK_USE_SYSTEM_KWIML=ON
    -DITK_USE_SYSTEM_MINC=OFF
    -DITK_USE_SYSTEM_PNG=ON
    -DITK_USE_SYSTEM_SWIG=ON
    -DITK_USE_SYSTEM_TIFF=ON
    -DITK_USE_SYSTEM_ZLIB=ON
    -DITK_WRAP_IMAGE_DIMS="2;3;4"
    -DITK_WRAP_PYTHON=ON
    -DITK_WRAP_complex_double=ON
    -DITK_WRAP_covariant_vector_double=ON
    -DITK_WRAP_double=ON
    -DITK_WRAP_rgb_unsigned_short=ON
    -DITK_WRAP_rgba_unsigned_short=ON
    -DITK_WRAP_signed_char=ON
    -DITK_WRAP_signed_long_long=ON
    -DITK_WRAP_unsigned_long_long=ON
    -DITK_WRAP_unsigned_short=ON
    -DITK_WRAP_vector_double=ON
    -DModule_ITKImageIO=ON
    -DModule_ITKMeshIO=ON
    -DModule_ITKTransformIO=ON
    -DModule_MorphologicalContourInterpolation=ON
)
_pkgname=ITK
cmake -B "build" -S "${_pkgname}" \
  ${cmake_opts[@]} \
  -DITK_USE_GPU=ON
make -C "build"
```

**Comment by tbirdso** [comment]
Resolved in #3444.
